src/nanoemoji/glue_together.py

```diff
@@ -84,19 +91,101 @@ def _copy_svg(target: ttLib.TTFont, donor: ttLib.TTFont):
     target["SVG "] = donor["SVG "]
 
 
+def _cbdt_data_and_sizes(ttfont: ttLib.TTFont) -> Mapping[str, CbdtGlyphInfo]:
+    data = {}
+    for strike_data in ttfont["CBDT"].strikeData:
+        data.update(strike_data)
+
+    sizes = {}
+    for strike in ttfont["CBLC"].strikes:
+        for sub_table in strike.indexSubTables:
+            for name, (start, end) in zip(sub_table.names, sub_table.locations):
+                sizes[name] = end - start
+
+    assert data.keys() == sizes.keys(), f"{data.keys()} != {sizes.keys()}"
+
+    return {
+        glyph_name: CbdtGlyphInfo(data[glyph_name], sizes[glyph_name])
+        for glyph_name in data
+    }
+
+
+def _copy_cbdt(target: ttLib.TTFont, donor: ttLib.TTFont):
+    cbdt_glyph_info = _cbdt_data_and_sizes(donor)
+
+    # reorder the bitmap table to match the targets glyph order
+    # we only support square bitmaps so the strikes are all the same
+    # other than glyph names so we can just construct a new
+    # order that matches that of target
+    donor_order = list(cbdt_glyph_info.keys())
+    # confirm our core assumption that we successfully held glyph names stable
+    if not set(donor_order) <= set(target.getGlyphOrder()):
+        raise ValueError("Donor glyph names do not exist in target")
+    new_order = sorted(donor_order, key=target.getGlyphID)
+
+    # now we know the desired order, reshard into runs
+    # TODO duplicative of make_cbdt_table
+    # take the first strike as a template, then wipe out strikes and data
+    # so we can build it up again in a potentially different glyph order
+    strike_template = donor["CBLC"].strikes[0]
+    clbc_index_template = strike_template.indexSubTables[0]
+    strike_template.indexSubTables = []
+    cblc = donor["CBLC"]
+    cblc.strikes = []
+    cbdt = donor["CBDT"]
+    cbdt.strikeData = []
+
+    data_offset = bitmap_tables.CBDT_HEADER_SIZE
+    while new_order:
+        # grab the next run w/consecutive gids
+        min_gid = target.getGlyphID(new_order[0])
+        end = 1
+        while (
+            len(new_order) > end
+            and target.getGlyphID(new_order[end])
+            == target.getGlyphID(new_order[end - 1]) + 1
+        ):
+            end += 1
+        glyph_run = new_order[:end]
+        new_order = new_order[end:]
+        max_gid = target.getGlyphID(glyph_run[-1])
+
+        strike = copy.deepcopy(strike_template)
+        strike.bitmapSizeTable.min_gid = min_gid
+        strike.bitmapSizeTable.max_gid = max_gid
+
+        max_width = max(cbdt_glyph_info[gn].data.metrics.width for gn in glyph_run)
+        strike.bitmapSizeTable.hori.widthMax = max_width
```

Comment:
in bitmap_tables.py you define widthMax as 
```python
line_metrics.widthMax = max(d.metrics.Advance for d in data.values())
```

you should do the same here I think, no?
metrics.width is the width of the bitmap, not the "advance" width.. So confusing, I know.
Also below for vertical

Reply:
Good catch, fixed